src/common/services/UserService/UserService.ts

```diff
@@ -24,6 +25,35 @@ const createUser = async (values: CreateUserType): Promise<Response> => {
   });
 };
 
+const readUsers = async (params: ReadUsersQueryType): Promise<ReadUsersResponseType> => {
+  let fetchQuery = `${import.meta.env.VITE_API_URL}/admin/users`;
+  if (params?.query) {
+    fetchQuery = addQueryParams(fetchQuery, params.query);
+  }
+  try {
+    return await fetch(fetchQuery)
+      .then((response) => {
+        return response.json();
+      })
+      .then((data: ReadUsersResponseType) => {
+        return data;
+      });
+  } catch (error) {
+    throw new FetchError("Couldn't read users");
+  }
+};
+
 export const useCreateUserMutation = () => {
   return useMutation(createUser);
 };
+
+export const useReadUsersMutation = (handleInputData: (data: UserDTO[], totalElements: number | undefined) => void) => {
```

Comment:
maybe better use `?: number` we do not need to make it `undefined` as this can be some source of confusion. For example: "Why do I Can pass a parameter that is undefined?" we can make it optional and skip this kind of questions